File: db4oj/core/src/com/db4o/events/CommitEventArgs.java
/* Copyright (C) 2007 db4objects Inc. http://www.db4o.com */

package com.db4o.events;

import com.db4o.ext.*;
import com.db4o.internal.*;

/**
 * Arguments for commit time related events.
 * 
 * @see EventRegistry
 */
public class CommitEventArgs extends TransactionalEventArgs {
	
	private final CallbackObjectInfoCollections _collections;

	public CommitEventArgs(Transaction transaction, CallbackObjectInfoCollections collections) {
		super(transaction);
		_collections = collections;
	}
	
	/**
	 * Returns a iteration
	 * 
	 * @sharpen.property
	 */
	public ObjectInfoCollection added() {
		return _collections.added;
	}
	
	/**
	 * @sharpen.property
	 */
	public ObjectInfoCollection deleted() {
		return _collections.deleted;
	}

	/**
	 * @sharpen.property
	 */
	public ObjectInfoCollection updated() {
		return _collections.updated;
	}
}

File: db4on/core/src/core/com/db4o/events/CommitEventArgs.cs
namespace com.db4o.events
{
	/// <summary>Arguments for commit time related events.</summary>
	/// <remarks>Arguments for commit time related events.</remarks>
	/// <seealso cref="com.db4o.events.EventRegistry">com.db4o.events.EventRegistry</seealso>
	public class CommitEventArgs : System.EventArgs
	{
		private readonly com.db4o.ext.ObjectInfoCollection _added;

		private readonly com.db4o.ext.ObjectInfoCollection _deleted;

		private readonly com.db4o.ext.ObjectInfoCollection _updated;

		public CommitEventArgs(com.db4o.ext.ObjectInfoCollection added, com.db4o.ext.ObjectInfoCollection
			 deleted, com.db4o.ext.ObjectInfoCollection updated)
		{
			_added = added;
			_deleted = deleted;
			_updated = updated;
		}

		/// <summary>Returns a iteration</summary>
		public virtual com.db4o.ext.ObjectInfoCollection Added
		{
			get
			{
				return _added;
			}
		}

		public virtual com.db4o.ext.ObjectInfoCollection Deleted
		{
			get
			{
				return _deleted;
			}
		}

		public virtual com.db4o.ext.ObjectInfoCollection Updated
		{
			get
			{
				return _updated;
			}
		}
	}
}
